day6/day6.rb

```diff
@@ -0,0 +1,71 @@
+require_relative './input'
+
+class Day6
+  def initialize(input)
+
+    @competition = Competition.new(input)
+  end
+
+  def part_one
+    @competition.number_of_wins_product
+  end
+
+  def part_two
+    @competition.number_of_wins_product
+  end
+end
+
+ class Competition
+  def initialize(input)
+
+    document = input.split("\n").reject{ |line| line == ""}
+    times = document[0].split(":")[1].split(" ")
+    distances = document[1].split(":")[1].split(" ")
+
+    @races = []
+    times.each_index do |i|
+      @races.push Race.new([times[i].to_i, distances[i].to_i])
+    end
+
+  end
+
+  def number_of_wins_product
+    total = nil
+    @races.each do |race|
+      if total != nil
+        total = total * race.number_of_wins
+      else
+        total = race.number_of_wins
+      end
+    end
+    total
+  end
+ end
+
+ class Race
+
+  def initialize input
+    @time ||= input[0]
+    @distance ||= input[1]
```

Comment:
its not possible for `@time` and `@distance` here to be anything but nil and initialize can only be called once. You can just use regular assignment here `=` not `||=`

I'd also not pass input like this. Better to explicitly pass arguments 

```ruby
def initialize(time, distance)
  @time = time
  @distance = distance
end
```